pkgs/top-level/all-packages.nix

```diff
@@ -1919,6 +1919,16 @@ in
     pinentry = if stdenv.isDarwin then pinentry_mac else pinentry;
   };
   gnupg = gnupg21;
+  # suitable to verify signatures non-interactively
+  gnupg-minimal = callPackage ../tools/security/gnupg/21.nix {
```

Comment:
Using `.override` is preferable here, like [curlFull](https://github.com/volth/nixpkgs/blob/dfb39da36e65c674c98fab2055c0e6377ae5239d/pkgs/top-level/all-packages.nix#L1332-L1336).